### PR TITLE
Bump tenacity to 6.2

### PR DESCRIPTION
**Pull request by kaxil**
There is no reason to limit tenacity. One of my upcoming PRs (or a commit in scheduler-ha PR) will use a new feature available in the newer version.

One of the reasons for creating a separate PR for this is so that [constraints](https://github.com/apache/airflow/tree/constraints-master) are updated and when I create new PR (or merge to scheduler-ha PR) it will run against a newer version of tenacity because of the upgraded constraints, without it, it will error.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by ryw — APPROVED** [review]
+1

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
@kaxil -> I would like to check if this works - do not merge it please before I take a look. I have a feeling we have some problem with upgrading constraints and I would like to have closer look tomorrow on it.

**Review by potiuk — CHANGES_REQUESTED** [review]
Just want to take a look at this before it gets merged.

**Comment by ashb** [comment]
why did we limit it in the first place? something must have broken with 5.1?

**Comment by potiuk** [comment]
I think @kaxil want to update constraints. But I am afraid it will not work this way. I will take a look at this thing tomorrow.

**Comment by kaxil** [comment]
No no my intention is to increase the limit too, constraints is just the by-product which will help in the next OR that will use the feature from latest version

**Comment by kaxil** [comment]
Also, no there is no reason it was pinned, apparently I did that one too: https://github.com/apache/airflow/commit/926aa1ba3a6ee8eec7b8efca2cc7f03d347134b7

**Comment by kaxil** [comment]
@potiuk  I don't see what changes yoh requested?

**Comment by kaxil** [comment]
Just to be clear, I want to update requirements not just constraints. Constraints is the by-product. I am familiar with the current CI problem regarding constraints.

**Review by ashb — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**
